bcos-executor/src/executive/ExecutiveFactory.cpp

```diff
@@ -87,21 +96,30 @@ std::shared_ptr<bcos::precompiled::Precompiled> ExecutiveFactory::getPrecompiled
 }
 
 std::shared_ptr<TransactionExecutive> ShardingExecutiveFactory::build(
-    const std::string& _contractAddress, int64_t contextID, int64_t seq, bool useCoroutine)
+    const std::string& _contractAddress, int64_t contextID, int64_t seq, bool useCoroutine, bool isSerialFlow)
 {
+    std::shared_ptr<TransactionExecutive> executive;
     if (useCoroutine)
     {
         auto needUsePromise = m_isTiKVStorage;  // tikv storage need to use promise executive
-        auto executive = std::make_shared<ShardingTransactionExecutive>(m_blockContext,
+        executive = std::make_shared<ShardingTransactionExecutive>(m_blockContext,
             _contractAddress, contextID, seq, m_gasInjector, m_poolForPromiseWait, needUsePromise);
-        setParams(executive);
-        return executive;
+
     }
     else
     {
-        auto executive = std::make_shared<TransactionExecutive>(
-            m_blockContext, _contractAddress, contextID, seq, m_gasInjector);
-        setParams(executive);
-        return executive;
+        if(isSerialFlow)
+        {
+            executive = std::make_shared<TransactionExecutiveSerial>(
+                m_blockContext, _contractAddress, contextID, seq, m_gasInjector);
+        }
+        else
+        {
+            executive = std::make_shared<TransactionExecutive>(
+                m_blockContext, _contractAddress, contextID, seq, m_gasInjector);
+        }
```

Comment:
sharding这里可以先不考虑，这个类用来build分片的executive
```suggestion
        assert(!isSerialFlow);
        executive = std::make_shared<TransactionExecutive>(
                m_blockContext, _contractAddress, contextID, seq, m_gasInjector);
)
```